src/dex/maverick-v2/maverick-v2.ts

```diff
@@ -0,0 +1,436 @@
+  readonly hasConstantPriceLargeAmounts = false;
+  readonly needWrapNative = true;
+
+  readonly isFeeOnTransferSupported = false;
+
+  public static dexKeysWithNetwork: { key: string; networks: Network[] }[] =
+    getDexKeysWithNetwork(MaverickV2Config);
+
+  logger: Logger;
+
+  public static erc20Interface = new Interface(ERC20ABI);
+
+  constructor(
+    readonly network: Network,
+    readonly dexKey: string,
+    readonly dexHelper: IDexHelper,
+    protected adapters = {},
+    protected config = MaverickV2Config[dexKey][network],
+    protected maverickV2Iface = new Interface(MaverickV2PoolABI),
+  ) {
+    super(dexHelper, dexKey);
+    this.logger = dexHelper.getLogger(dexKey);
+  }
+
+  async initializePricing(blockNumber: number) {
+    await this.setupEventPools(blockNumber);
+  }
+
+  async setupEventPools(blockNumber: number) {
+    const pools = (await this._queryPoolsAPI(SUBGRAPH_TIMEOUT))?.pools;
+
+    await Promise.all(
+      pools?.map(async (pool: any) => {
+        const eventPool = new MaverickV2EventPool(
+          this.dexKey,
+          this.network,
+          this.dexHelper,
+          this.logger,
+          {
+            address: pool.tokenA.address,
+            symbol: pool.tokenA.symbol,
+            decimals: pool.tokenA.decimals,
+          },
+          {
+            address: pool.tokenB.address,
+            symbol: pool.tokenB.symbol,
+            decimals: pool.tokenB.decimals,
+          },
+          pool.fee * 1e4 * 1e14,
+          pool.feeB * 1e4 * 1e14,
+          pool.tickSpacing,
+          0,
+          pool.lookback,
+          pool.lowerTick,
+          pool.id,
+          this.config.poolLensAddress,
+        );
+        const onChainState = await eventPool.generateState(blockNumber);
+        if (blockNumber) {
+          eventPool.setState(onChainState, blockNumber);
+          this.dexHelper.blockManager.subscribeToLogs(
+            eventPool,
+            eventPool.addressesSubscribed,
+            blockNumber,
+          );
+        }
+        this.pools[eventPool.address] = eventPool;
+      }) || [],
+    );
+  }
+
+  getAdapters(side: SwapSide): { name: string; index: number }[] | null {
+    return null;
+  }
+
+  async getPoolIdentifiers(
+    srcToken: Token,
+    destToken: Token,
+    side: SwapSide,
+    blockNumber: number,
+  ): Promise<string[]> {
+    const from = this.dexHelper.config.wrapETH(srcToken);
+    const to = this.dexHelper.config.wrapETH(destToken);
+    if (from.address.toLowerCase() === to.address.toLowerCase()) {
+      return [];
+    }
+
+    const pools = await this.getPools(from, to);
+    return pools.map(
+      (pool: any) => `${this.dexKey}_${pool.address.toLowerCase()}`,
+    );
+  }
+
+  async getPools(srcToken: Token, destToken: Token) {
+    return Object.values(this.pools).filter((pool: MaverickV2EventPool) => {
+      return (
+        (pool.tokenA.address.toLowerCase() == srcToken.address.toLowerCase() ||
```

Comment:
@Wertusser could we use strict equality ===  everywhere within this integration ?